draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -335,20 +332,84 @@ hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
+        As illustrated in <xref target="fig5"/>, the AS path is considered a route leak (i.e., valley free violation <xref target="RFC7908"/>) if for
+        some i and j (j &ge; i+1), the hop AS(i) to AS(i+1) is either a P2C or p2p while
+        the hop from AS(j+1) to AS(j) is also a P2C or p2p.
+        Note that the AS(j+1) to AS(j) hop is checked in a direction opposite to the Update flow direction.  
+        There are four combinations here and they are illustrated as A, B, C, and D in <xref target="fig5"/>.
+        For the upstream path, AS(N+1) to AS(N) obviously known to be P2C or P2P,
+        hence the value of j can always be set equal to N.
+        So, for determining if an upstream path is Invalid (route leak), all that is needed is to find an i (1 &lt; i &lt; N-1)
+        for which AS(i+1) is not a provider of AS(i) per AS(i)'s ASPA.
+        This makes the algorithm for upstream paths quite simple.
+        For the downstream path, j is obviously not equal to N, and hence both i and j have to be searched for
+        (in the range between 1 and N-1) by checking the AS hops from left to right and right to left
+        with the help of the ASPAs. The hop-check function (<xref target="pair-validation"/>) is utilized here.
+        The algorithm for downstream paths is therefore more complex and requires more computations compared to the algorithm for upstream paths.
+        Thus, it is prudent to use separate algorithms for the verification of upstream and downstream paths (<xref target="Upflow"/> and <xref target="Downflow"/>, respectively).
+      </t>
+      <t>
+        <figure anchor="fig5" align="left" suppress-title="false" pn="figure-2">
+          <name slugifiedName="route-leak">Illustration of the four route leak types (valley-free violations) . </name>
+          <artwork align="left" name="" type="" alt="">
+<![CDATA[
+                 <-------- Update flow direction ------
+ (A) .... AS(j+1)              ........                AS(i) ....
+             \_____(P2C)___                 ____(P2C)___/
+                           \               /
+                           \/             \/
+                          A(j)          AS(i+1)
+
+ (B) .... AS(j+1)---(P2P)--->A(j)   ........           AS(i) ....
+                                             ___(P2C)___/
+                                            /
+                                           \/
+                                           AS(i+1)
+
+ (C) .... AS(j+1)            ......   AS(i+1)<---P2P---AS(i) ....
+             \_____P2C_____          
+                           \            
+                           \/           
+                          A(j)
+
+ (D) .... AS(j+1)---(P2P)--->A(j) ..... AS(i+1)<---P2P---AS(i) ....
+                               
+]]>
+</artwork>
+        </figure>
       </t>
+    </section>
```

Comment:
I think you are referring to Section 7.2. It can be improved as follows. The text should upfront refer to RFC7908 where the four key types of route leaks (valley-free violations) are defined in Secs 3.1 through 3.4. Basically, a route leak happens if the update is sent from a provider to a customer or between lateral peers on one hop and subsequently sent from a customer to a provider or between lateral peers on another hop. The two hops where this happens need not be adjacent.